### ISSUE TITLE
[bug]: Can't find use-mobile hook in shadcn doc

### ISSUE THREAD

**Issue by ardevik** [shadcn-ui/ui#6133]
### Describe the bug

Because restricted firewall in my company, I can't use the cli for adding components because it point to external registry, so I'm using manual installation for sidebar component and adding requirement one by one. Until I want add the @/components/hooks/use-mobile but can't find the example

### Affected component/components

Sidebar

### How to reproduce

1. Go to shadcn website
2. Add Sidebar with Manual Installation 

### System Info

```bash
Chrome
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by akshay-nm** [comment]
the import generated in the sidebar.tsx (/components/sidebar.tsx) is 
```
import { useIsMobile } from "@/components/hooks/use-mobile"
```

this hook is not installed via init command. Any ideas where I get this?

**Comment by i-am-henri** [comment]
@akshay-nm  You can get this hook via the repo:
![{0BE091C1-FD1A-47EC-A3E4-FF99D8F60474}](https://github.com/user-attachments/assets/d960fea1-cb80-450e-a1a9-5f5045e12f27)

This is the hook:

```ts
import * as React from "react"

const MOBILE_BREAKPOINT = 768

export function useIsMobile() {
  const [isMobile, setIsMobile] = React.useState<boolean | undefined>(undefined)

  React.useEffect(() => {
    const mql = window.matchMedia(`(max-width: ${MOBILE_BREAKPOINT - 1}px)`)
    const onChange = () => {
      setIsMobile(window.innerWidth < MOBILE_BREAKPOINT)
    }
    mql.addEventListener("change", onChange)
    setIsMobile(window.innerWidth < MOBILE_BREAKPOINT)
    return () => mql.removeEventListener("change", onChange)
  }, [])

  return !!isMobile
}
```

**Comment by akshay-nm** [comment]
i tried`npx shadcn@latest add sidebar-01` as suggested here: https://ui.shadcn.com/blocks/sidebar

It worked, but one issue is, the shadcn add command overwrites the lib/utils.tsx (not related to this issue).

**Comment by DemaPy** [comment]
[PR](https://github.com/shadcn-ui/ui/pull/6203) created to close issue.

**Comment by DemaPy** [comment]
@ardevik pls close issue once PR will be accepted.

@akshay-nm  consider to create new issue for such task. It will be CLI development.

**Comment by DemaPy** [comment]
@akshay-nm  Also it looks strange that utils.tsx file has been overwritten. Because default behavior for add command is: ```overwrite: false```

<img width="433" alt="image" src="https://github.com/user-attachments/assets/b8d1826e-e2ef-4c0b-9717-2d6ec61315c6" />

And there is also check in file ```update-files.ts```
<img width="652" alt="image" src="https://github.com/user-attachments/assets/25542bc4-d069-4b59-8f4d-b2f4b7d95531" />

So, please make sure to test it again and if it happend create new issue.

**Comment by j127** [comment]
It seems like this works with the Radix version but not with the Base UI version in TanStack Start. It only installs the use-mobile file when using the Radix version.

## Reproduction

Create a project:

```bash
bunx --bun shadcn@latest create --preset "https://ui.shadcn.com/init?base=base&style=vega&baseColor=neutral&theme=neutral&iconLibrary=lucide&font=inter&menuAccent=subtle&menuColor=default&radius=default&template=start" --template start
```

Add the sidebar:

```bash
bunx --bun shadcn@latest add sidebar
```

The `use-mobile.ts` file isn't there:

```text
✔ Checking registry.
✔ Created 4 files:
  - src/components/ui/skeleton.tsx
  - src/components/ui/tooltip.tsx
  - src/components/ui/sheet.tsx
  - src/components/ui/sidebar.tsx
ℹ Skipped 3 files: (files might be identical, use --overwrite to overwrite)
  - src/components/ui/button.tsx
  - src/components/ui/input.tsx
  - src/components/ui/separator.tsx
```

Try again with the default command that uses Radix:

```bash
bun create @tanstack/start@latest --tailwind --add-ons shadcn
bunx --bun shadcn@latest add sidebar
```

and the `use-mobile.ts` file is created:

```text
✔ Checking registry.
✔ Updating CSS variables in src/styles.css
✔ Installing dependencies.
✔ Created 8 files:
  - src/components/ui/button.tsx
  - src/components/ui/separator.tsx
  - src/components/ui/sheet.tsx
  - src/components/ui/tooltip.tsx
  - src/components/ui/input.tsx
  - src/hooks/use-mobile.ts
  - src/components/ui/skeleton.tsx
  - src/components/ui/sidebar.tsx
```

**Comment by bennik88** [comment]
Same issue with Base UI

**Comment by ShayokhShorfuddin** [comment]
Yup just faced same issue with BaseUI.

<img width="1381" height="111" alt="Image" src="https://github.com/user-attachments/assets/a485f35a-cda7-40d9-ab9f-f1f6d3b14912" />

**Comment by ErikAbele** [comment]
Yes, looks like `use-mobile` is missing in the dependency list for `base/ui/sidebar.tsx` in the registry index: https://github.com/shadcn-ui/ui/blob/main/apps/v4/registry/bases/__index__.tsx

But it can be easily installed explicitly until someone gets around fixing it:

`npx shadcn@latest add use-mobile`
